be/src/formats/csv/varbinary_converter.cpp

```diff
@@ -54,44 +53,37 @@ Status VarBinaryConverter::write_quoted_string(OutputStream* os, const Column& c
 }
 
 bool VarBinaryConverter::read_string(Column* column, const Slice& s, const Options& options) const {
-    int max_size = 0;
+    int max_size = -1;
     if (options.type_desc != nullptr) {
         max_size = options.type_desc->len;
     }
 
     char* data_ptr = static_cast<char*>(s.data);
     int start = StringParser::skip_leading_whitespace(data_ptr, s.size);
-    int len = s.size - start;
-    // hex's length should be 2x.
-    if (len % 2 != 0) {
-        LOG(WARNING) << "Column [" << column->get_name() << "]'s length invalid:" << s.to_string();
-        return false;
+    if (start == s.size) {
+        column->append_default();
+        return true;
     }
 
-    // hex's length should not greater than MAX_VARCHAR_LENGTH.
-    int hex_len = len / 2;
-    if (config::enable_check_string_lengths &&
-        ((hex_len > TypeDescriptor::MAX_VARCHAR_LENGTH) || (max_size > 0 && hex_len > max_size))) {
-        LOG(WARNING) << "Column [" << column->get_name() << "]'s length exceed max varbinary length.";
-        return false;
-    }
+    auto buf = new char[s.size + 3];
+    DeferOp defer([&]() { delete[] buf; });
+
+    int r = base64_decode2(s.data + start, s.size, buf);
+    char* data = buf;
+    int len = r;
 
-    // check slice is valid
-    for (int i = start; i < s.size; i++) {
-        if (LIKELY((s[i] >= '0' && s[i] <= '9') || (s[i] >= 'A' && s[i] <= 'F') || (s[i] >= 'a' && s[i] <= 'f'))) {
-            continue;
-        } else {
-            LOG(WARNING) << "Invalid input's not a legal hex-encoded value:" << s.to_string();
-            return false;
-        }
+    if (r == -1) { // fallback if decode failed
```

Comment:
If the target csv's varbinary was not encoded by base64, it will fail every time, which will affect performance a lot.

Reply:
that's why Trino is so slow :)

Reply:
Given some random csv file, how do you know if it's base64-encoded?